visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -47,6 +46,29 @@ export class NodeDecorator {
 			})
 	}
 
+	public static doesExclusionResultInEmptyMap(map: CodeMapNode, blacklist: BlacklistItem[]): boolean {
+		let excludedNodes = 0
+		const flattened = ignore()
+		const excluded = ignore()
+
+		for (const item of blacklist) {
+			const path = CodeMapHelper.transformPath(item.path)
+			item.type === BlacklistType.flatten ? flattened.add(path) : excluded.add(path)
```

Comment:
Please always use `if (foo) { func(bar) } else { otherFunc(bar) }` instead of the ternary operator when calling functions. Maybe use a lint rule for it :)

Reply:
Done. I couldn't find a rule that checks for non-return calls in ternary though. :(

Reply:
Seems like you missed this one? :-) let's check for such rule later on. Seems like `flattened` is not used here at all though?

Reply:
```ts
	public static decorateMapWithBlacklist(map: CodeMapNode, blacklist: BlacklistItem[]) {
		const flattened = ignore()
		const excluded = ignore()

		for (const item of blacklist) {
			const path = CodeMapHelper.transformPath(item.path)
			if (item.type === BlacklistType.flatten) {
				flattened.add(path)
			} else {
				excluded.add(path)
			}
		}

		hierarchy(map)
			.descendants()
			.map(node => {
				const path = CodeMapHelper.transformPath(node.data.path)
				node.data.isFlattened = flattened.ignores(path)
				node.data.isExcluded = excluded.ignores(path)
			})
	}
```

It has been adressed and flattened is required to set the flattened attribute as well. Otherwise flattening a building would not work. :)

Reply:
OH. I just realized there are actually two methods who used that ternary. You marked another one. my bad